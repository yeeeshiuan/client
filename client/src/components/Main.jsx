import React, { Component } from 'react';
import axios from 'axios';

class Main extends Component {
    constructor() {
        super();
        this.state = {
            data : {},
        };

        this.getData = this.getData.bind(this);
    }

    componentDidMount() {
    // if there is no channel name there, get the top stream channel
        this.getData("thermalPowerGenerationCO2");
    }

    getData(apiKey) {

        const options = {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json',
            },
            url: `${process.env.REACT_APP_API_DOMAIN_NAME_URL}/base/api/` + apiKey,
        };

        axios(options)
        .then(response => { 

            this.setState({ data:response.data });
            console.log(response.data);
        })
    }

    render() {
        return (
                <div>
                    <p>This is my App.(Main)</p>
                    
                </div>
        )
    }
}

export default Main;
