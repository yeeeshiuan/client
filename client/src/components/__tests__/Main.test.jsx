import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import Main from '../Main';


test('Main renders without crashing', () => {
  const wrapper = shallow(<Main/>);
});

test('Main contain a p tag', () => {
  const wrapper = shallow(<Main />);
  const p = wrapper.find('p');
  expect(p.length).toBe(1);
  expect(p.get(0).props.children).toBe('This is my App.(Main)');
});

test('Main will call componentDidMount when mounted', () => {
  const onDidMount = jest.fn();
  Main.prototype.componentDidMount = onDidMount;
  const wrapper = mount(<Router><Main/></Router>);
  expect(onDidMount).toHaveBeenCalledTimes(1)
});

test('Main renders a snapshot properly', () => {
  const tree = renderer.create(<Main/>).toJSON();
  expect(tree).toMatchSnapshot();
});
