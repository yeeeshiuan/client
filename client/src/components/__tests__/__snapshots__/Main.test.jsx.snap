// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders a snapshot properly 1`] = `
<div>
  <p>
    This is my App.(Main)
  </p>
  <div
    style={
      Object {
        "height": 200,
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
